oci/private/index.bzl

```diff
@@ -0,0 +1,77 @@
+"Implementation details for oci_index rule"
```

Comment:
is the naming `index` a common name for a multi-arch OCI compatible image?

Reply:
index is a technical name for multi-arch image as it's made of list of manifests. maybe `oci_image_index` is a better name? 
https://github.com/opencontainers/image-spec/blob/main/image-index.md

Reply:
yup. lets switch it to `oci_image_index`. that looks like name in the spec as well https://specs.opencontainers.org/image-spec/image-index/?v=v1.0.1

Reply:
agreed.